public/components/trace_analytics/components/traces/trace_view_helpers.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import moment from 'moment';
+
+export function normalizePayload(parsed: any): any[] {
+  if (Array.isArray(parsed)) {
+    return parsed;
+  }
+  if (parsed.hits && Array.isArray(parsed.hits.hits)) {
+    return parsed.hits.hits;
+  }
+  return [];
+}
+
+export function getOverviewFields(parsed: any, mode: string) {
+  const hits = normalizePayload(parsed);
+  if (hits.length === 0) return null;
+  const firstSpan = hits[0]._source;
+  if (!firstSpan) return null;
+
+  if (mode === 'jaeger') {
+    // In Jaeger mode, we assume fields:
+    //   startTime: a number in microseconds
+    //   duration: a number in microseconds
+    const lastUpdated = Number(firstSpan.startTime) + Number(firstSpan.duration) / 1000;
+    return {
+      trace_id: firstSpan.traceID || '-',
+      trace_group: firstSpan.operationName || '-',
+      last_updated: moment(lastUpdated).format('MM/DD/YYYY HH:mm:ss'),
+      latency: firstSpan.duration ? `${(Number(firstSpan.duration) / 1000).toFixed(2)} ms` : 'N/A',
+      error_count: firstSpan.tag && firstSpan.tag.error ? 1 : 0,
+    };
+  } else {
+    // Data Prepper mode: try to use traceGroupFields, but if they're null, fall back to top-level values.
+    let computedLatency: number | null = null;
+    if (firstSpan.traceGroupFields && firstSpan.traceGroupFields.durationInNanos != null) {
+      computedLatency = Number(firstSpan.traceGroupFields.durationInNanos) / 1e6;
+    } else if (firstSpan.startTime && firstSpan.endTime) {
+      computedLatency = moment(firstSpan.endTime).diff(
+        moment(firstSpan.startTime),
+        'milliseconds',
+        true
+      );
+    }
```

Comment:
Based on the community feedback, should we be showing a note or a caution icon, saying the values fall back to top-level values when the traceGroupFields is null?

Reply:
<img width="861" alt="LatencyWarning" src="https://github.com/user-attachments/assets/9d2b59c5-2e4a-4a56-a2b6-9d73cf1d50fb" />
Added icon for if a fallback value is used
"Latency may not be accurate due to missing or unexpected duration data."